src/sys/unix.rs

```diff
@@ -538,6 +570,35 @@ impl Socket {
         Ok((n as usize, addr))
     }
 
+    #[cfg(not(target_os = "redox"))]
+    pub fn recv_vectored(&self, bufs: &mut [IoSliceMut<'_>], flags: c_int) -> io::Result<usize> {
+        let mut msg: libc::msghdr = unsafe { std::mem::zeroed() };
+        msg.msg_iov = bufs.as_mut_ptr().cast();
+        msg.msg_iovlen = bufs.len().min(max_iov()) as _;
+
+        let n = syscall!(recvmsg(self.fd, &mut msg as *mut _, flags))?;
+        Ok(n as usize)
+    }
+
+    #[cfg(not(target_os = "redox"))]
+    pub fn recv_from_vectored(
+        &self,
+        bufs: &mut [IoSliceMut<'_>],
+        flags: c_int,
+    ) -> io::Result<(usize, SockAddr)> {
+        let mut storage: libc::sockaddr_storage = unsafe { mem::zeroed() };
+        let mut msg: libc::msghdr = unsafe { std::mem::zeroed() };
```

Comment:
Nit: same as above for the `libc::msghdr`.

Reply:
Done :)